escpos/config.py

```diff
@@ -0,0 +1,89 @@
+""" ESC/POS configuration manager.
+
+This module contains the implentations of abstract base class :py:class:`Config`.
+
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import os
+import appdirs
+import yaml
+
+from . import printer
+from . import exceptions
+
+class Config(object):
+    """  Configuration handler class.
+
+    This class loads configuration from a default or specificed directory. It
+    can create your defined printer and return it to you.
+    """
+    _app_name = 'python-escpos'
+    _config_file = 'config.yaml'
+
+    def __init__(self):
+        self._has_loaded = False
+        self._printer = None
+
+        self._printer_name = None
+        self._printer_config = None
+
+    def load(self, config_path=None):
+        """ Load and parse the configuration file using pyyaml
+
+        :param config_path: An optional file path, file handle, or byte string
+            for the configuration file.
+
+        """
+        if not config_path:
+            config_path = os.path.join(
+                appdirs.user_config_dir(self._app_name),
+                self._config_file
+            )
+
+        try:
+            if isinstance(config_path, str):
```

Comment:
Sorry, I have not seen that one before, but by importing unicode_literals I have found a problem:
In Python 2 the type of `config_path` is (then) unicode. That's why the config is not properly loaded and therefore the else-clause is taken. (I don't quite understand that one; why are you calling yaml.safe_load with config_path?)
The easiest fix to this type-check is to write `isinstance(config_path, (str, unicode))` instead of `isinstance(config_path, str)`. 

One further idea: Maybe you should throw an exception if the `printer`-section hasn't been found in the config. Otherwise the resulting error (1) will be rather unintuitive. This can however also be fixed later.

(1)
```
self._printer = getattr(printer, self._printer_name)(**self._printer_config)
TypeError: getattr(): attribute name must be string
```